src/seedu/addressbook/data/person/Address.java

```diff
@@ -1,18 +1,36 @@
 package seedu.addressbook.data.person;
 
+import java.util.HashMap;
+
 import seedu.addressbook.data.exception.IllegalValueException;
 
 /**
  * Represents a Person's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ * Guarantees: the individual parts are immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String ADDRESS_SPLIT_WORD = ", ";
+    public static final String EXAMPLE = AddressBlock.EXAMPLE + ADDRESS_SPLIT_WORD 
+                                    + AddressStreet.EXAMPLE + ADDRESS_SPLIT_WORD 
+                                    + AddressUnit.EXAMPLE + ADDRESS_SPLIT_WORD
+                                    + AddressPostalCode.EXAMPLE;
+    
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Address should have four components split by comma";
+    // Note that the address parts will also check validity in their own classes
+    public static final String ADDRESS_VALIDATION_REGEX = ".+,.+,.+,.+";
 
+    public enum EnumAddressPart {
+        BLOCK(0), STREET(1), UNIT(2), POSTAL_CODE(3);
+        
+        public final int id;
+        EnumAddressPart(int id) {
+            this.id = id;
+          }
+    }
     public final String value;
+
```

Comment:
By conflicts, do you mean that I should not save the value of the address at all and instead return the values of the individual objects (the block, postal code etc) together?